pkg/controller/build/build_controller.go

```diff
@@ -161,37 +162,58 @@ func (r *ReconcileBuild) Reconcile(request reconcile.Request) (reconcile.Result,
 		return reconcile.Result{}, err
 	}
 
+	target := instance.DeepCopy()
+	targetLog := rlog.ForBuild(target)
+
+	if target.Status.Phase == v1alpha1.BuildPhaseNone || target.Status.Phase == v1alpha1.BuildPhaseWaitingForPlatform {
+		pl, err := platform.GetCurrentPlatform(ctx, r.client, target.Namespace)
+		switch {
+		case err != nil:
+			target.Status.Phase = v1alpha1.BuildPhaseError
+			target.Status.Failure = v1alpha1.NewErrorFailure(err)
+		case pl.Status.Phase != v1alpha1.IntegrationPlatformPhaseReady:
+			target.Status.Phase = v1alpha1.BuildPhaseWaitingForPlatform
+		default:
+			target.Status.Phase = v1alpha1.BuildPhaseInitialization
+		}
+
+		if instance.Status.Phase != target.Status.Phase {
+			err = r.update(ctx, target)
+			if err != nil {
```

Comment:
We may want to factorise the conflict error handling in the `update` method as well.